### PR TITLE
New agent configuration and specific hardware files

### PR DESCRIPTION
**Pull request by AAndrisa**

### CODE REVIEW

File: tools/scripts/build_projects.py [hunk 1]
```diff
@@ -380,4 +381,4 @@ def main():
 main()
 
 if ERR != 0:
-	exit("Error occured: %d" % ERR)
+	exit("Error occured: %d" % ERR)
```
Comment (amiclaus) [inline tools/scripts/build_projects.py:394]: nit: new line

File: build_projects.yml [hunk 2]
```diff
@@ -23,18 +26,174 @@ jobs:
...
+    inputs:
+      targetPath: '$(Build.Repository.LocalPath)/$(RELEASE_DIR)'
+      artifact: '$(ARTIFACT_NAME)_ADuCM3029'
+      publishLocation: 'pipeline'
+  - task: PublishPipelineArtifact@1
+    condition: true
+    inputs:
+      targetPath: '$(Build.Repository.LocalPath)/$(LOG_DIR)'
+      artifact: 'logs_ADuCM3029_$(System.JobId)'
+      publishLocation: 'pipeline'
```
Comment (amiclaus) [inline build_projects.yml:207]: nit: new line

File: tools/scripts/download_files.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        folder_git_sha = timestamp_folder_info.split('[')[2].split(']')[0]
+                        folder_commit_date = timestamp_folder_info.split('[')[1].split(']')[0]
+                        print("Hardware " + hardware + " found on next timestamp " + timestamp_folder + " with next properties git sha: " + \
+                            str(folder_git_sha) + " commit date: " + str(folder_commit_date))
+                    else:
+                        print("Hardware " + hardware + " found on next timestamp " + timestamp_folder)
+                    FOUND = True
+                    break
+    if FOUND is False:
+        print("Project " + hardware + " was not found on server")
```
Comment (amiclaus) [inline tools/scripts/download_files.py:65]: nit: new line

**Review by amiclaus — COMMENTED** [review]
is it possible to cherry-pick from master branch the commits for the new agent configuration and afterwards add separate commit on top that handles the hardware files from specific branch? I think we want to keep the  history of next_stable branch similar the the master branch, as much as possible.

**Comment by AAndrisa** [comment]
V2: The commits from master for new agent configuration were added by cherry-picking and above them the new commits.

**Review by SRaus — APPROVED** [review]

**Review by SRaus — APPROVED** [review]

**Pull request merged**